Port IPC::FormDataReference to the new IPC serialization format

cdumez opened the pull request:
#### 63f2089fb62fc77995e76546329006288840b840
<pre>
Port IPC::FormDataReference to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=264895">https://bugs.webkit.org/show_bug.cgi?id=264895</a>

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/IPC/FormDataReference.h:
(IPC::FormDataReference::data const):
(IPC::FormDataReference::takeData):
(IPC::FormDataReference::FormDataReference):
(IPC::FormDataReference::sandboxExtensionHandles const):
(IPC::FormDataReference::encode const): Deleted.
(IPC::FormDataReference::decode): Deleted.
* Source/WebKit/Platform/IPC/FormDataReference.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/270801@main">https://commits.webkit.org/270801@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/244545259cb4dd4e1988fa68f4b7bdc48795f050

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26411 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5024 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27670 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28568 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24174 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6827 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2452 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24158 "37 flakes 144 failures") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26671 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3877 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22709 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3432 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/3464 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29107 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24091 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24084 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29743 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3482 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1708 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27638 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4909 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3963 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3417 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3807 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Platform/IPC/FormDataReference.h:
```diff
@@ -40,53 +40,37 @@ class FormDataReference {
+        return { };
 
-        return FormDataReference { WTFMove(*formData) };
+    Vector<WebKit::SandboxExtension::Handle> sandboxExtensionHandles;
```
reserveInitialCapacity?

cdumez replied:
I'll compactMap().

achristensen07 commented on Source/WebKit/Platform/IPC/FormDataReference.h:
```diff
@@ -40,53 +40,37 @@ class FormDataReference {
+inline FormDataReference::FormDataReference(RefPtr<WebCore::FormData>&& data, Vector<WebKit::SandboxExtension::Handle>&& sandboxExtensionHandles)
+    : m_data(WTFMove(data))
+{
+    WebKit::SandboxExtension::consumePermanently(sandboxExtensionHandles);
```
WTFMove?

cdumez replied:
Wouldn't have any benefit at the moment?
```
bool consumePermanently(const Vector<Handle>&);
```

cdumez committed:
Port IPC::FormDataReference to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=264895

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/IPC/FormDataReference.h:
(IPC::FormDataReference::data const):
(IPC::FormDataReference::takeData):
(IPC::FormDataReference::FormDataReference):
(IPC::FormDataReference::sandboxExtensionHandles const):
(IPC::FormDataReference::encode const): Deleted.
(IPC::FormDataReference::decode): Deleted.
* Source/WebKit/Platform/IPC/FormDataReference.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/270801@main
